tutorials/python-api.md

```diff
@@ -0,0 +1,108 @@
+# LitGPT High-level Python API
+
+This is a work-in-progress draft for a high-level LitGPT Pyhon API.
+
+&nbsp;
+## Model loading & saving
+
+The `LLM.load` command loads an `llm` object, which contains both the model object (a PyTorch module) and a preprocessor.
+
+```python
+from litgpt import LLM
+
+llm = LLM.load(
+    source="url | local_path",
+    # high-level user only needs to care about those:
+    memory_reduction="none | medium | strong"
+    # advanced options for technical users:
+    hub="hf | local | other"
+    quantize="bnb.nf4",
+    precision="bf16-true",
```

Comment:
Would the memory_reduction argument be a replacement for these?

Reply:
Yes, exactly. For non-technical users, `memory_reduction` would be the preferred choice. But we should still have the option of manually specifying `quantize="bnb.nf4",` and `precision="bf16-true",` for experienced users. We just have to add warning/message that `memory_reduction` and these settings are mutually exclusive.

Reply:
It feels like even a novice user will end up searching what medium|strong means and will sea that it's some combo of quantization/precision.
In other words I don't see a point of this.